Add config_error method to fetch error on config

tombruijn opened the pull request:
To help out with #1389, add a way to get the error that occurred on loading the config file available. This way people can check in the app if an error occurred and reraise it or exit the app if necessary.

We don't want to add the behavior of raising the config error on start by default, because the gem should never raise an error.

I decide not to put it behind a config option, because the only way to do that is with an environment variable. This also gives more flexibility for anyone who want to add any kind of custom behavior based on the presence of an error.

unflxw commented:
If I recall correctly, we added an environment variable that could be turned on to make AppSignal fail to start if the configuration file couldn't be parsed.

Maybe we can add an environment variable (and/or a `start!` method?) that can raise an exception if AppSignal fails to start?

tombruijn commented:
> If I recall correctly, we added an environment variable that could be turned on to make AppSignal fail to start if the configuration file couldn't be parsed.

That env var was to not start AppSignal if the `config/appsignal.yml` file would run into an error. It would silently fail (we'd still log it) to start.
(The original behavior was to ignore the config file if an error occurred.)

We removed this in Ruby gem 4 and made it the default behavior.

> Maybe we can add an environment variable (and/or a start! method?) that can raise an exception if AppSignal fails to start?

The `start!` method wouldn't be a good idea, because not always can the app call `Appsignal.start(!)` because we do (in the Railtie).

We could add a `Appsignal.raise_if_not_started!` kind of method that just does `raise Appsignal.config_error if Appsignal.config_error?` as implemented in this PR.

I don't love adding another env var for this (I was very happy to remove the other one), that changes the start behavior.

tombruijn commented:
Added a `Appsignal.check_if_started!` method in #1395
